src/style/style_layer/fill_extrusion_style_layer.js

```diff
@@ -232,12 +231,12 @@ function projectExtrusion2D(geometry: Array<Array<Point>>, zBase: number, zTop:
             const baseX = sX + baseXZ;
             const baseY = sY + baseYZ;
             const baseZ = sZ + baseZZ;
-            const baseW = sW + baseWZ;
+            const baseW = Math.max(sW + baseWZ, 0.00001);
```

Comment:
A comment or link to this issue might be helpful since the reason for this line may not be obvious.

Also, can `baseW` be negative but not close to zero? Just checking since it's not clear to me what this would compute if that were the case. For example: `Math.sign(sW + baseWZ) * Math.max(Math.abs(sW + baseWZ), 0.00001);`.

Reply:
(To at least be clear if I'm going to be pedantic, because it's homogeneous, the result would be exactly identical if all of the components were simply negated—unless the clamping didn't permit negative but otherwise equivalent w. Just wanted to be certain that it's an upstream guarantee that this wouldn't happen.)

Reply:
> A comment or link to this issue might be helpful since the reason for this line may not be obvious.

There is no issue for this. Discovered it while porting various QRF pr's to gl-native and needed a test, so also fixed it here. The PR and query tests should serve as plenty of documentation.

> Also, can baseW be negative but not close to zero?

Yes, that is actually a major part of the issue

>  For example, to avoid zero without otherwise changing the magnitude or sign of the number

It's basically clamping, which serves us well in this case as we don't need the magnitude when w < 0.